server/aws-lsp-codewhisperer/src/shared/utils.ts

```diff
@@ -533,15 +533,28 @@ export async function listFilesWithGitignore(directory: string): Promise<string[
         }
     }
 
-    const absolutePaths = await fg(['**/*'], {
+    const absolutePaths: string[] = []
+    let fileCount = 0
+    const MAX_FILES = 100_000
+
+    const stream = fg.stream(['**/*'], {
         cwd: directory,
         dot: true,
         ignore: ignorePatterns,
-        onlyFiles: false,
+        onlyFiles: true,
         followSymbolicLinks: false,
         absolute: true,
     })
-    return absolutePaths.slice(0, 500_000)
+
+    for await (const entry of stream) {
```

Comment:
Do we need the for loop, can we just change the value in the slice above to 100K?

Reply:
No. This is for early termination of fast glob. Without for loop it iterate first.